### PR TITLE
Curve decoders improvement

### PR DESCRIPTION
**Pull request by yabirgb**
Closes #(issue_number)

## Checklist

- [ ] The PR modified the frontend, and updated the [user guide](https://github.com/rotki/rotki/blob/develop/docs/usage_guide.rst) to reflect the changes.

### CODE REVIEW

**Comment by yabirgb** [comment]
Need to check the failing tests. I'm happy to have made the tests for the decoders

File: rotkehlchen/chain/ethereum/modules/curve/decoder.py [hunk 1]
```diff
@@ -124,6 +141,17 @@ def _decode_curve_deposit_events(
...
+            elif (  # deposit give asset
+                event.event_type == HistoryEventType.SPEND and
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.location_label == user_address and
+                tx_log.address in self.curve_pools
+            ):
+                event.event_type = HistoryEventType.DEPOSIT
+                event.event_subtype = HistoryEventSubType.DEPOSIT_ASSET
+                event.counterparty = CPT_CURVE
+                event.notes = f'Deposit {event.balance.amount} {event.asset.symbol} in curve pool'  # noqa: E501
```
Comment (LefterisJP) [inline rotkehlchen/chain/ethereum/modules/curve/decoder.py:153]: Can you maybe get the curve pool name here somehow? Not a must. Just a question/good to have.

File: rotkehlchen/chain/ethereum/modules/curve/decoder.py [hunk 2]
```diff
@@ -152,6 +190,27 @@ def _decode_curve_events(  # pylint: disable=no-self-use
...
+    ) -> bool:
+        source_address = hex_or_bytes_to_address(tx_log.topics[1])
+        to_address = hex_or_bytes_to_address(tx_log.topics[2])
+        if (  # deposit give asset
+            event.event_type == HistoryEventType.RECEIVE and
+            event.event_subtype == HistoryEventSubType.NONE and
+            source_address == CURVE_Y_DEPOSIT and
+            transaction.from_address == to_address
+        ):
+            event.event_subtype = HistoryEventSubType.REMOVE_ASSET
```
Comment (LefterisJP) [inline rotkehlchen/chain/ethereum/modules/curve/decoder.py:209]: Okay so here we may have a problem.

We need to figure out what we want to call deposit and withdrawal in pools.

As you can see we treat it the same in say aave, compound.

So deposit is:
```
                event.event_type = HistoryEventType.DEPOSIT
                event.event_subtype = HistoryEventSubType.DEPOSIT_ASSET
```

Receive the wrapped/pool token is:

```
                event.event_type = HistoryEventType.RECEIVE
                event.event_subtype = HistoryEventSubType.RECEIVE_WRAPPED
```


Then withdrawing from the pool is. Step 1(send back pool token):

```
                event.event_type = HistoryEventType.SPEND
                event.event_subtype = HistoryEventSubType.RETURN_WRAPPED
```

step 2: get back the actual token:

```
                event.event_type = HistoryEventType.WITHDRAWAL
                event.event_subtype = HistoryEventSubType.REMOVE_ASSET
```

This is how aave/compound pools are.

You are doing it different here. This is why they don't show well in the frontend. There is no mapping for this.
The mappings of all possible combos are here:

https://github.com/rotki/rotki/blob/4e50a76b62362714e589b2bebf8370d5a6e24295/frontend/app/src/store/history/consts.ts#L289

I would say it still makes sense to treat them like we do with aave and compound? Right?

**Review by LefterisJP — COMMENTED** [review]
Two comments.

File: rotkehlchen/chain/ethereum/modules/curve/decoder.py (hunk 1 above)
Comment (yabirgb) [reply rotkehlchen/chain/ethereum/modules/curve/decoder.py]: I've added it where is possible to know exactly the contract where they were deposited/withdrawn

File: rotkehlchen/chain/ethereum/modules/curve/decoder.py (hunk 2 above)
Comment (yabirgb) [reply rotkehlchen/chain/ethereum/modules/curve/decoder.py:212]: Absolutely. I've fixed that one and checked in the frontend

**Review by LefterisJP — APPROVED** [review]
LGTM. If there is more stuff to fix we do it later.

And by the way by curve pool name I meant name such as stETH/WETH etc. Not an address. But we can see if this is possible later.

**Pull request merged**